cloud/amazon/iam_server_certificate_facts.py

```diff
@@ -0,0 +1,168 @@
+expiration:
+    description: The date and time this server certificate will expire, in ISO 8601 format.
+    returned: success
+    type: str
+    sample: "2017-06-15T12:00:00+00:00"
+upload_date:
+    description: The date and time this server certificate was uploaded, in ISO 8601 format.
+    returned: success
+    type: str
+    sample: "2015-04-25T00:36:40+00:00"
+'''
+
+
+try:
+    import boto3
+    import botocore.exceptions
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_server_cert(iam, name=None):
+    """Retrieve the attributes of a server certificate if it exists
+    Args:
+        iam (botocore.client.IAM): The boto3 iam instance.
+
+    Kwargs:
+        name (str): The name of the server certificate.
+
+    Basic Usage:
+        >>> import boto3
+        >>> iam = boto3.client('iam')
+        >>> name = "server-cert-name"
+        >>> results = get_server_cert(iam, name)
+        [
+            {
+                "upload_date": "2015-04-25T00:36:40+00:00",
+                "server_certificate_id": "ADWAJXWTZAXIPIMQHMJPO",
+                "certificate_body": "-----BEGIN CERTIFICATE-----\nbunch of random data\n-----END CERTIFICATE-----",
+                "server_certificate_name": "server-cert-name",
+                "expiration": "2017-06-15T12:00:00+00:00",
+                "path": "/",
+                "arn": "arn:aws:iam::911277865346:server-certificate/server-cert-name"
+            }
+        ]
+    """
+    results = []
+    try:
+        server_cert = iam.get_server_certificate(ServerCertificateName=name)['ServerCertificate']
+        cert_md = server_cert['ServerCertificateMetadata']
+        cert_data = {
+            'certificate_body': server_cert['CertificateBody'],
+            'server_certificate_id': cert_md['ServerCertificateId'],
+            'server_certificate_name': cert_md['ServerCertificateName'],
+            'arn': cert_md['Arn'],
+            'path': cert_md['Path'],
+            'expiration': cert_md['Expiration'].isoformat(),
+            'upload_date': cert_md['UploadDate'].isoformat(),
+        }
+        results.append(cert_data)
+
+    except botocore.exceptions.ClientError:
+        pass
+    return results
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(required=True, type='str'),
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec,)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        iam = boto3_conn(module, conn_type='client', resource='iam', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Boto3 Client Error - " + str(e.msg))
+
+    cert_name = module.params.get('name')
+    results = get_server_cert(iam, cert_name)
+    module.exit_json(results=results)
```

Comment:
This returns an array containing the object, instead can you have it a dictionary with the name of the cert as the key? For example:

```
{
    "mycert": { <all the data currently returned, just not in an array> }
}
```

The reason I'd prefer this is so, in the future, the facts module could have a default behavior of listing *all* certs if a name isn't provided, and they'd all be in the returned dictionary nested under their name.